### PR TITLE
#167646812 create nav bar

### PR DESCRIPTION
**Pull request by timi-codes**
#### What does this PR do?
- create app header
- Configure sass to work with web-pack and jest
- Implemented the modularised project structure for store
#### Link to Pivotal Story
[#167646812](https://www.pivotaltracker.com/story/show/167646812)

### CODE REVIEW

**Review by Onyimatics — APPROVED** [review]
Good job @timi-codes . Thanks for fixing the errors in the test and the other things you fixed.

File: src/components/NavBar/index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  const [menuOpen, setMenuOpen] = React.useState(false);
+
+  const toggleMenu = () => {
+    setMenuOpen(!menuOpen);
+  };
+
+  return (
+    <nav className="wrapper flex flex-row items-center justify-between pt-2 pb-2 md:pl-10 md:pr-10 sm:pl-4 sm:pr-4">
+      <div className="logo flex items-center">
+        <a
```
Comment (segunolalive) [inline src/components/NavBar/index.js]: Use React-router `Link` tag.

**Review by segunolalive — COMMENTED** [review]
attend to the comments below. Good work.

File: __test__/App.spec.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          </Switch>
+        </BrowserRouter>
+      </Provider>
+    );
+    const wrapper = mount(comp);
+
+    expect(wrapper.find('Home')).toBeTruthy();
+  });
+  it('should work fine on App', () => {
+    const wrapper = shallow(<App />);
```
Comment (olorunwalawrence) [inline __test__/App.spec.js]: kindly change the shallow method to mount as the TTL suggested

**Review by kechyy — APPROVED** [review]
@Good Job @timi-codes . I think I like the quality of your test. 

It looks good to me.

**Review by codexempire — APPROVED** [review]
👍 😄 🥇

File: src/components/NavBar/index.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  );
+};
+
+MenuIcon.propTypes = {
+  isLoggedIn: PropTypes.bool.isRequired,
+  menuOpen: PropTypes.bool.isRequired,
+  toggleMenu: PropTypes.func.isRequired
+};
+
+const AuthButtons = ({ isLoggedIn }) => {
```
Comment (segunolalive) [inline src/components/NavBar/index.js]:
```js
function AuthButtons ({ isLoggedIn = false }) => {
   if(isLoggedIn) { return null; }
   return ( // the rest of your code);
}

function AuthButtons ({ show = false }) => {
   return ( 
      show ? <your fragment here> : null
   );

  usage: <AuthButtons show={!isLoggedIn} />
}


```

**Review by segunolalive — CHANGES_REQUESTED** [review]
Good job

File: tailwind.config.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// tailwind.config.js
+module.exports = {
+  theme: {
+    screens: {
+      sm: { min: '375px' },
+      // => @media (max-width: 639px) { ... }
```
Comment (segunolalive) [inline tailwind.config.js]: remove dead code (comments)

**Review by segunolalive — COMMENTED** [review]
Looking good

File: package.json [hunk 5]
```diff
@@ -20,7 +20,8 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "lint-staged"
+      "pre-commit": "lint-staged",
+      "pre-push": "npm test"
```
Comment (tohbay) [inline package.json:24]: Thank you @timi-codes for adding the `pre-push` command. 👍

File: src/assets/icons/index.js [hunk 6]
```diff
@@ -1,7 +1,21 @@
...
+  SearchIcon,
+  NotificationIcon,
+  ClearIcon,
+  AvatarIcon,
+  FacebookIcon,
+  InstagramIcon,
+  LinkedInIcon,
+  YoutubeIcon,
+  TwitterIcon
+};
```
Comment (tohbay) [inline src/assets/icons/index.js:21]: Good job here for removing the dead code. 👍

File: tailwind.config.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+// tailwind.config.js
```
Comment (tohbay) [inline tailwind.config.js]: Remove this dead code on Line 1

**Review by tohbay — COMMENTED** [review]
Great efforts have been demonstrated in this PR. 
Kudos @timi-codes 👍 
Is there a review link where we can test out this PR?

File: src/components/NavBar/NavBar.scss [hunk 8]
```diff
@@ -0,0 +1,91 @@
+@import '../../../public/styles/colors.scss';
+@import '../../../public/styles/responsive.scss';
+
+.logo {
+  width: 163px;
+  height: 55px;
```
Comment (tohbay) [inline src/components/NavBar/NavBar.scss:5]: I think you should use relative sizes like `em`, `rem` and `%`

**Review by olorunwalawrence — CHANGES_REQUESTED** [review]
image

I think the login and get started buttons should not be included in the mobile view of the header since they are being catered for in this PR: https://github.com/andela/ah-kifaru-frontend/pull/15 , kindly remove them, and I think the logo image should not be fixed kindly use some relative sizes e.g rem, em or percentage, the hamburger needs some right padding from  the edge of the view.

**Pull request merged**